src/Mapper/DatabaseMapper.php

```diff
@@ -110,11 +106,8 @@ public function queueCreate($entity, Node $node, State $state): ContextCarrierIn
             $this->primaryColumn
         );
 
-        if (!array_key_exists($this->primaryKey, $columns)) {
-            $insert->forward(Insert::INSERT_ID, $state, $this->primaryKey);
-        } else {
-            $insert->forward($this->primaryKey, $state, $this->primaryKey);
-        }
+        $key = isset($columns[$this->primaryKey]) ? $this->primaryKey : Insert::INSERT_ID;
```

Comment:
I'm pretty sure the array_key_exists was for a purpose. It will react to the cases where PK not defined vs when it's null.

Reply:
Hello @wolfy-j ,

in  line 102: https://github.com/cycle/orm/blob/master/src/Mapper/DatabaseMapper.php#L102
the `$this->primaryKey` $columns key is removed if the value is `null`, so isset() and array_key_exists() will return the same value, isset call being a little faster for php < 7.4

Reply:
That’s true, thank you